### DISCUSSION TITLE
Load entities from world folder

### DISCUSSION THREAD

**Discussion by ghost** [Minestom/Minestom#741] [Help] [answered]
So, I created world with a lot of entities and it doesn't load in Minestom. I tried to edit Minestom's source code and `getEntities` in `ChunkColumn` returns `[]`

**Comment by emortaldev** [comment]
The AnvilLoader does not read entities yet, Minestom also cannot serialize or deserialize entities yet, and that would have to be done first

**Comment by jglrxavpok** [comment]
Issue was closed on Hephaistos repo https://github.com/jglrxavpok/Hephaistos/issues/18

**Comment by jglrxavpok** [comment]
Format changed in 1.17, entities are now stored inside entities/ instead of region/
https://minecraft.fandom.com/wiki/Entity_format
Will require changes on Hephaistos' side

**Comment by jglrxavpok** [comment, marked as answer]
Possible with the current version of Hephaistos, but still a bit involved for the moment:

Here's my answer on the Hephaistos issue

> It is possible to load entities with the current version of Hephaistos, it is just a bit involved: Load your region file from the entities/ folder as usual. Instead of querying the ChunkColumn, get the NBT data directly via `RegionFile#getChunkData(chunkX, chunkZ)`. Then you should be able to access the `Entities` (mind the capital E) NBTCompound array. From there, you can load your entities as you wish.
> 
> There will be a better/cleaner/faster way of doing things at some point, but there's quite some changes required.
> 
> Here's a small snippet:
> 
> ```java
> String filename = "myworld/entities/r.0.0.mca";
> RegionFile f = new RegionFile(new RandomAccessFile(filename, "r"), 0, 0);
> NBTList<NBTCompound> list = f.getChunkData(0,0).getList("Entities");
> System.out.println(list.toSNBT());
> ```
